### ISSUE TITLE
Config parsing file structure is problematic + circular imports

### ISSUE THREAD

**Issue by yngve-sk** [equinor/ert#5196]
In general, the code tends to export all members of a package, when it would perhaps be possible to export less members. This exporting of all members is problematic and introduces potential circular dependencies.

One example of this: The newer ert config parser and related logic lives under `_c_wrappers/enkf`, and it isn't really a c wrapper anymore. It still has some dependencies tied towards the old parser, which has great potential for introducing circular imports.

* We have `_c_wrappers/config` where `ConfigValidationError` lives, this is imported from various places
* In `_c_wrappers/enkf/config` we have more c wrappers for configuration parsing
* To achieve compatibility between old and new parsers, the new parser `ert_config` has to import the `ConfigValidationError` from `_c_wrappers/config_parser.py`
* It should be possible to move all raising of config validation errors AWAY FROM `_c_wrappers/config_parser.py`, so that the new parser in `ert_config.py` and `lark_parser` etc does not have to import error logic from the old parser.

Possible solutions/TODO's
* Remove all error raising / error classes (specifically `ConfigValidationError`) from `_c_wrappers/config_parser.py` so that it lives alongside the new parser. Suggestion: Either raise another type, or indicate errors in another way (for example via a list)
* Move `lark_parser`, `ert_config` and related files to `src/ert/config`, decouple as much as possible from old parser logic.

TLDR @eivindjahren: Just make it simple

**Comment by yngve-sk** [comment]
Resolved by https://github.com/equinor/ert/pull/5212
